ui/admin/tests/acceptance/credential-store/credentials/delete-test.js

```diff
@@ -44,73 +46,183 @@ module(
       assert.strictEqual(currentURL(), urls.credentials);
-      assert.strictEqual(getCredentialsCount(), credentialsCount - 1);
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernamePasswordCredentialCount - 1
+      );
     });
 
-    test('cannot delete a credential store without proper authorization', async function (assert) {
+    test('can delete username & key pair credential', async function (assert) {
+      assert.expect(2);
+      const usernameKeyPairCredentialCount =
+        getUsernameKeyPairCredentialCount();
+      await visit(urls.usernameKeyPairCredential);
+      await click('.rose-layout-page-actions .rose-dropdown-button-danger');
+      assert.strictEqual(currentURL(), urls.credentials);
+      assert.strictEqual(
+        getUsernameKeyPairCredentialCount(),
+        usernameKeyPairCredentialCount - 1
+      );
+    });
+
+    test('cannot delete a username & password credential without proper authorization', async function (assert) {
       assert.expect(3);
-      const credentialsCount = getCredentialsCount();
-      instances.credential.authorized_actions =
-        instances.credential.authorized_actions.filter(
+      const usernamePasswordCredentialCount =
+        getUsernamePasswordCredentialCount();
+      instances.usernamePasswordCredential.authorized_actions =
+        instances.usernamePasswordCredential.authorized_actions.filter(
           (item) => item !== 'delete'
         );
-      await visit(urls.credential);
-      assert.strictEqual(currentURL(), urls.credential);
+      await visit(urls.usernamePasswordCredential);
+      assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
       assert.notOk(
         find('.rose-layout-page-actions .rose-dropdown-button-danger')
       );
-      assert.strictEqual(getCredentialsCount(), credentialsCount);
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernamePasswordCredentialCount
+      );
     });
 
-    test('can accept delete credential via dialog', async function (assert) {
+    test('cannot delete a username & key pair credential without proper authorization', async function (assert) {
+      assert.expect(3);
+      const usernameKeyPairCredentialCount =
+        getUsernameKeyPairCredentialCount();
+      instances.usernameKeyPairCredential.authorized_actions =
+        instances.usernameKeyPairCredential.authorized_actions.filter(
+          (item) => item !== 'delete'
+        );
+      await visit(urls.usernameKeyPairCredential);
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
+      assert.notOk(
+        find('.rose-layout-page-actions .rose-dropdown-button-danger')
+      );
+      assert.strictEqual(
+        getUsernamePasswordCredentialCount(),
+        usernameKeyPairCredentialCount
+      );
+    });
+
+    test('can accept delete username & password credential via dialog', async function (assert) {
```

Comment:
What is the difference between this test and `can delete username & key pair credential`?

Reply:
Since this test just mocks out the dialog box, the regular delete button is used. So, the main difference is that the dialog mock is used in this test. I see why this test is confusing, so I agree with actually clicking through the buttons instead of using mocks. I will change this.